Interactive component creation fails if disconnected from the cluster(offline)

/kind bug
## What versions of software are you using?

**Operating System:**
All

**Output of `odo version`:**
odo v2.4.3 (12b79aaa5)

## How did you run odo exactly?
1. `odo create`

## Actual behavior
Fails at retrieving namespace.
```shell
➜  odo create               
Devfile Object Creation
? Which devfile component type do you wish to create  [Use arrows to move, enter
? Which devfile component type do you wish to create  [Use arrows to move, enter
? Which devfile component type do you wish to create  [Use arrows to move, enter
? Which devfile component type do you wish to create  [Use arrows to move, enter
? Which devfile component type do you wish to create go
? What do you wish to name the new devfile component go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x130 pc=0x292c3a6]

goroutine 1 [running]:
github.com/redhat-developer/odo/pkg/odo/cli/component.InteractiveCreateMethod.FetchDevfileAndCreateComponent(0xc00045e2c0, 0xc0003e5680, 0x47744a0, 0x0, 0x0, 0x2da1840, 0xc00074df50)
        /home/pvala/go/src/github.com/redhat-developer/odo/pkg/odo/cli/component/create_methods.go:65 +0x5a6
github.com/redhat-developer/odo/pkg/odo/cli/component.(*CreateOptions).Complete(0xc00045e2c0, 0xc000385e30, 0x6, 0xc0003e5680, 0x47744a0, 0x0, 0x0, 0x0, 0x0)
        /home/pvala/go/src/github.com/redhat-developer/odo/pkg/odo/cli/component/create.go:194 +0x451
github.com/redhat-developer/odo/pkg/odo/genericclioptions.GenericRun(0x359c2c0, 0xc00045e2c0, 0xc0003e5680, 0x47744a0, 0x0, 0x0)
        /home/pvala/go/src/github.com/redhat-developer/odo/pkg/odo/genericclioptions/runnable.go:85 +0x3c4
github.com/redhat-developer/odo/pkg/odo/cli/component.NewCmdCreate.func1(0xc0003e5680, 0x47744a0, 0x0, 0x0)
        /home/pvala/go/src/github.com/redhat-developer/odo/pkg/odo/cli/component/create.go:356 +0x5e
github.com/spf13/cobra.(*Command).execute(0xc0003e5680, 0x47744a0, 0x0, 0x0, 0xc0003e5680, 0x47744a0)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:856 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0009b9400, 0xc000058150, 0x2ea1e80, 0x47744a0)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:897
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:890
main.main()
        /home/pvala/go/src/github.com/redhat-developer/odo/cmd/odo/odo.go:70 +0x426
```
## Expected behavior
Should successfully create component.

## Any logs, error output, etc?
Errors indicate that the panic is caused by uninitialized value of KClient, see https://github.com/redhat-developer/odo/blob/main/pkg/odo/cli/component/create_methods.go#L65.

--- a/pkg/odo/cli/component/create_methods.go
+++ b/pkg/odo/cli/component/create_methods.go
@@ -61,17 +61,21 @@ func (icm InteractiveCreateMethod) FetchDevfileAndCreateComponent(co *CreateOpti
 		if err != nil {
 			return err
 		}
-	} else if co.KClient != nil {
-		// Get the current project if no --project is passed, but --now is passed
-		componentNamespace = ui.EnterDevfileComponentProject(co.KClient.GetCurrentProjectName())
 	} else {
-		var client kclient.ClientInterface
-		client, err = kclient.New()
-		// Continue if an error is encountered
-		if err == nil && client != nil {
-			// Get the current project if no --project is passed and using offline context
-			componentNamespace = ui.EnterDevfileComponentProject(client.GetCurrentProjectName())
+		var currentNamespace string
+		if co.KClient != nil {
+			// Get the current project if no --project is passed, but --now is passed
+			currentNamespace = co.KClient.GetCurrentProjectName()
+		} else {
+			var client kclient.ClientInterface
+			client, err = kclient.New()
+			// if err != nil, currentNamespace will be an empty string
+			if err == nil && client != nil {
+				// Get the current project if no --project is passed and using offline context
+				currentNamespace = client.GetCurrentProjectName()
+			}
 		}
+		componentNamespace = ui.EnterDevfileComponentProject(currentNamespace)
 	}
 
 	co.devfileMetadata.componentType = componentType

--- a/pkg/odo/cli/component/create_methods.go
+++ b/pkg/odo/cli/component/create_methods.go
@@ -2,6 +2,7 @@ package component
 
 import (
 	"fmt"
+	"github.com/redhat-developer/odo/pkg/kclient"
 	"io/ioutil"
 	"os"
 	"path/filepath"
@@ -60,9 +61,15 @@ func (icm InteractiveCreateMethod) FetchDevfileAndCreateComponent(co *CreateOpti
 		if err != nil {
 			return err
 		}
-	} else {
-		// if the user is logged in or if we have cluster information, display the default project
+	} else if co.KClient != nil {
+		// Get the current project if no --project is passed, but --now is passed
 		componentNamespace = ui.EnterDevfileComponentProject(co.KClient.GetCurrentProjectName())
+	} else {
+		client, _ := kclient.New()
+		if client != nil {
+			// Get the current project if no --project is passed and using offline context
+			componentNamespace = ui.EnterDevfileComponentProject(client.GetCurrentProjectName())
+		}
 	}
 
 	co.devfileMetadata.componentType = componentType

--- a/pkg/odo/cli/component/create_methods.go
+++ b/pkg/odo/cli/component/create_methods.go
@@ -65,8 +65,10 @@ func (icm InteractiveCreateMethod) FetchDevfileAndCreateComponent(co *CreateOpti
 		// Get the current project if no --project is passed, but --now is passed
 		componentNamespace = ui.EnterDevfileComponentProject(co.KClient.GetCurrentProjectName())
 	} else {
-		client, _ := kclient.New()
-		if client != nil {
+		var client kclient.ClientInterface
+		client, err = kclient.New()
+		// Continue if an error is encountered
+		if err == nil && client != nil {
 			// Get the current project if no --project is passed and using offline context
 			componentNamespace = ui.EnterDevfileComponentProject(client.GetCurrentProjectName())
 		}

